include/pmacc/mappings/kernel/AreaMapping.hpp

```diff
@@ -26,9 +26,34 @@
 #include "pmacc/mappings/kernel/AreaMappingMethods.hpp"
 #include "pmacc/types.hpp"
 
+#include <cstdint>
+
 namespace pmacc
 {
-    template<uint32_t areaType, class baseClass>
+    /** Mapping between block indices and supercells in the given area for alpaka kernels
+     *
+     * The mapping covers the case of supercell-level parallelism between alpaka blocks.
+     * The supercells can be processed concurrently in any order.
+     *
+     * This class makes a 1:1 mapping between supercells in the area and alpaka blocks.
+     * A kernel must be launched with exactly getGridDim() blocks.
+     * Each block must process a single supercell with index getSuperCellIndex(blockIndex).
+     * Implementation is optimized for the standard areas in AreaType.
+     *
+     * In-block parallelism is independent of this mapping and is done by a kernel.
+     * Naturally, this parallelism should also define block size, again independent of this mapping.
+     *
+     * This pattern is used in most kernels in particle-mesh codes.
+     * Two most common parallel patterns are:
+     *    - for particle or particle-grid operations:
```

Comment:
I think this (or something similar) would be useful in readthedocs for developers. As it explains the parallelism pattern in most our kernels and how it is expressed in the code via a mapper.